source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,866 @@
+object. This field SHOULD be private to ensure that a user can not modify it
+while a ``withTransaction`` call is in progress. Drivers that cannot make
+this field private MUST signal that the field should not be accessed or
+modified by users if there is an idiomatic way to do so in the language (e.g.
+underscore-prefixed variable names in Python) and MUST document that
+modification of the field can cause unintended correctness issues for
+applications. Drivers MUST document that the remaining timeout will not be
+applied to callback operations that do not use the ClientSession and that
+overridding ``timeoutMS`` for operations inside the provided callback will
+result in a client-side error. If the callback returns an error and the
+transaction must be aborted, drivers MUST refresh the ``timeoutMS`` value for
+the ``abortTransaction`` operation.
+
+If ``timeoutMS`` is not set, drivers MUST continue to exhibit the existing
+120 second timeout behavior. Drivers MUST NOT change existing implementations
+to use ``timeoutMS=120000`` for this case.
+
+See `withTransaction communicates timeoutMS via ClientSession`_ and
+`withTransaction refreshes the timeout for abortTransaction`_.
+
+GridFS API
+----------
+
+GridFS buckets MUST inherit ``timeoutMS`` from their parent MongoDatabase
+instance and all methods in the GridFS Bucket API MUST support the
+``timeoutMS`` option. For methods that create streams (e.g.
+``open_upload_stream``), the option MUST cap the lifetime of the entire
+stream. This MUST include the time taken by any operations executed during
+stream construction, reads/writes, and close/abort calls. For example, if a
+stream is created at time ``T``, the final ``close`` call on the stream MUST
+finish all blocking work before time ``T + timeoutMS``. Methods that interact
+with a user-provided stream (e.g. ``upload_from_stream``) MUST use
+``timeoutMS`` as the timeout for the entire upload/download operation. If the
+user-provided streams do not support timeouts, drivers MUST document that the
+timeout for these methods may be breached if calls to interact with the
+stream take longer than the remaining timeout. If ``timeoutMS`` is set, all
+cursors created for GridFS API operations MUST internally set the
+``timeoutMode`` option to ``CURSOR_LIFETIME``.
+
+See `GridFS streams behavior`_.
+
+RunCommand
+----------
+
+The behavior of ``runCommand`` is undefined if the provided command document
+includes a ``maxTimeMS`` field and the ``timeoutMS`` option is set. Drivers
+MUST document the behavior of ``runCommand`` for this case and MUST NOT
+attempt to check the command document for the presence of a ``maxTimeMS``
+field.
+
+See `runCommand behavior`_.
+
+Test Plan
+=========
+
+See the `README.rst
+<https://github.com/divjotarora/specifications/blob/csot-tests/source/client-side-operations-timeout/tests/README.rst>`__
+in the tests directory.
+
+Motivation for Change
```

Comment:
Also say something about supporting more than 1 retry for a given retryable operation.

Reply:
Added a small new paragraph for this.

Reply:
Perfect. Thanks.